centreon/src/Centreon/Domain/Common/Assertion/Assertion.php

```diff
@@ -298,6 +298,24 @@ public static function ipAddress(mixed $value, ?string $propertyPath = null): vo
         }
     }
 
+    /**
+     * @param object $value
+     * @param string $className
+     * @param string|null $propertyPath
+     *
+     * @throws \Assert\AssertionFailedException
+     */
+    public static function isInstanceOf(mixed $value, string $className, ?string $propertyPath = null): void
+    {
+        if (! ($value instanceof $className)) {
+            throw AssertionException::badInstanceOfObject(
+                is_object($value) ? get_class($value) : gettype($value),
```

Comment:
To me [get_debug_type](https://www.php.net/manual/fr/function.get-debug-type.php) is more suitable for error messages and debugging because it manages all the possibilities like anonymous classes + it is more consistent with the real type names (`bool` vs `boolean` for example).
```suggestion
                get_debug_type($value),
```